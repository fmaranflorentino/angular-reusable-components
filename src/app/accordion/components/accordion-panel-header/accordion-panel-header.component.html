<div class="accordion_panel__header__wrapper">
  <ng-content select="[headerTitle]"></ng-content>

  <ng-content select="[panelDescription]"></ng-content>

  <div class="custom_icon__container">
    <span class="custom_icon">
      <ng-content select="[headerIcon]"></ng-content>
    </span>
    <span class="material-icons toggle" *ngIf="!isHideToggle">
      keyboard_arrow_down
    </span>
  </div>
</div>
