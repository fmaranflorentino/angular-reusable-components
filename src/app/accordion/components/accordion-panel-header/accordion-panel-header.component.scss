.accordion_panel {
  &__header {
    margin-right: 10px;
    display: block;
    width: 100%;

    &__wrapper {
      display: flex;
      align-items: center;
      justify-content: space-between;

      .hideIcon {
        margin: 0;
      }
    }
  }
}
