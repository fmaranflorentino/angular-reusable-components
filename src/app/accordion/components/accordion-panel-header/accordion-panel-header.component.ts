import { Component, OnInit, HostBinding, ViewEncapsulation, Input } from '@angular/core';

@Component({
  selector: 'app-accordion-panel-header',
  templateUrl: './accordion-panel-header.component.html',
  styleUrls: ['./accordion-panel-header.component.scss'],
  encapsulation: ViewEncapsulation.None,
})
export class AccordionPanelHeaderComponent implements OnInit {
  @HostBinding('class.accordion_panel__header')
  accordionPanelHeader = true;

  @HostBinding('class.accordion_panel__header--hidetoggle')
  get hideToggleValue() {
    return this.isHideToggle === true;
  }

  @HostBinding('class.accordion_panel__header--toggleanime')
  get expandedValue() {
    return this.isExpanded === true;
  }

  @Input()
  set hideToggle(value: boolean) {
    this._hideToggle = value !== null && `${value}` !== 'false';
  }

  @Input()
  set expanded(value) {
    this._expanded = value;
  }

  get isHideToggle(): boolean {
    return this._hideToggle;
  }

  get isExpanded(): boolean {
    return this._expanded;
  }


  private _hideToggle: boolean;

  private _expanded: boolean;

  constructor() { }

  ngOnInit(): void {
  }

}
