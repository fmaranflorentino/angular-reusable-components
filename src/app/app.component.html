<section>
  <button customButton color="primary">
    primary
  </button>

  <button customButton color="secondary">
    secondary
  </button>

  <button customButton outline>
    outline
  </button>

  <button customButton outline>
    outline icon left
    <img src="/assets/send.svg" alt="" icon />
  </button>

  <button customButton color="primary" iconPosition="right">
    icon right
    <img src="/assets/send.svg" alt="" icon />
  </button>

  <button customButton disabled>
    disabled
  </button>
</section>
<hr />
<accordion>
  <accordion-panel>
    <h3 panel-header-title>Angular - Developer Experience</h3>
    <span accordion-panel-description
      >A meetup event for defining the future of developer experience.</span
    >
    <span class="material-icons" panel-header-icon>
      account_circle
    </span>

    <accordion-panel-content>content man</accordion-panel-content>
  </accordion-panel>

  <accordion-panel>
    <h3 panel-header-title>Angular - Developer Experience</h3>
    <span accordion-panel-description
      >A meetup event for defining the future of developer experience.</span
    >
    <span class="material-icons" panel-header-icon>
      account_circle
    </span>

    <accordion-panel-content>content man</accordion-panel-content>
  </accordion-panel>
</accordion>
