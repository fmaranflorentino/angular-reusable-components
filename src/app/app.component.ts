import { Component } from '@angular/core';
// import { testeServ } from './accordion/components/bottom-sheet/teste';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  // constructor(private teste: testeServ) {
  //   this.teste.open();
  // }
}
