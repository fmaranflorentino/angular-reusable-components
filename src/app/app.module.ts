import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { ButtonComponent } from './button/button.component';
import { AccordionComponent } from './accordion/accordion.component';
import { AccordionPanelComponent } from './accordion/components/accordion-panel/accordion-panel.component';
import { AccordionPanelHeaderComponent } from './accordion/components/accordion-panel-header/accordion-panel-header.component';
import { AccordionPanelDescriptionComponent } from './accordion/components/accordion-panel-description/accordion-panel-description.component';
import { CommonModule } from '@angular/common';
import { AccordionPanelContentComponent } from './accordion/components/accordion-panel-content/accordion-panel-content.component';
import { BottomSheatComponent } from './accordion/components/bottom-sheat/bottom-sheat.component';
import { BottomSheetComponent } from './accordion/components/bottom-sheet/bottom-sheet.component';
// import { testeServ } from './accordion/components/bottom-sheet/teste';
// import { OVERLAY_PROVIDERS } from '@angular/cdk/overlay';
import { PanelHeaderTitleComponent } from './accordion/components/accordion-panel-header/components/panel-header-title/panel-header-title.component';

@NgModule({
  declarations: [
    AppComponent,
    ButtonComponent,
    AccordionComponent,
    AccordionPanelComponent,
    AccordionPanelHeaderComponent,
    AccordionPanelDescriptionComponent,
    AccordionPanelContentComponent,
    BottomSheatComponent,
    BottomSheetComponent,
    PanelHeaderTitleComponent,
  ],
  imports: [BrowserModule, BrowserAnimationsModule, CommonModule],
  providers: [
    // testeServ,
    // OVERLAY_PROVIDERS,
  ],
  bootstrap: [AppComponent],
  exports: [
    ButtonComponent,
    AccordionComponent,
    AccordionPanelComponent,
    AccordionPanelHeaderComponent,
    AccordionPanelDescriptionComponent,
    AccordionPanelContentComponent,
    BottomSheatComponent,
    BottomSheetComponent,
    PanelHeaderTitleComponent,
  ],
  entryComponents: [BottomSheetComponent],
})
export class AppModule {}
