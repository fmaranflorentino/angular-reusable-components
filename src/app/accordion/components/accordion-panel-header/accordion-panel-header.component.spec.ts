import { SpectatorHost, createHostFactory } from '@ngneat/spectator/jest';
import { axe } from 'jest-axe';

import { AppModule } from '../../../app.module';
import { AccordionPanelHeaderComponent } from './accordion-panel-header.component';
import { APP_BASE_HREF } from '@angular/common';

const cases = {
  simple: `<app-accordion-panel-header></app-accordion-panel-header>`,
  expanded: `
  <app-accordion-panel-header [expanded]="true"></app-accordion-panel-header>
  `,
  hiddeToggle: `
  <app-accordion-panel-header [hideToggle]="true"></app-accordion-panel-header>
  `,
  title: `
  <app-accordion-panel-header>
    <ng-container headerTitle>
      <ng-content select="[panel-header-title]"></ng-content>
    </ng-container>
  </app-accordion-panel-header>`,
  description: `
  <app-accordion-panel-header>
    <span panelDescription>
      <ng-content select="[accordion-panel-description]"></ng-content>
    </span>
  </app-accordion-panel-header>`,
  customIcon: `
  <app-accordion-panel-header>
    <ng-container headerIcon>
      <ng-content select="[panel-header-icon]"></ng-content>
    </ng-container>
  </app-accordion-panel-header>`,
  complex: `
  <app-accordion-panel-header>
    <ng-container headerTitle>
      <ng-content select="[panel-header-title]"></ng-content>
    </ng-container>

    <span panelDescription>
      <ng-content select="[accordion-panel-description]"></ng-content>
    </span>

    <ng-container headerIcon>
      <ng-content select="[panel-header-icon]"></ng-content>
    </ng-container>
  </app-accordion-panel-header>`
};

describe('AccordionPanelHeaderComponent', () => {
  let subject: SpectatorHost<AccordionPanelHeaderComponent>;
  const createHost = createHostFactory({
    component: AccordionPanelHeaderComponent,
    declareComponent: false,
    imports: [AppModule],
    providers: [{ provide: APP_BASE_HREF, useValue: '/' }],
  });

  it('should create instance', () => {
    subject = createHost(cases.simple);
    expect(subject.component).toBeTruthy();
  });

  it('should create and match raw component snapshot', () => {
    subject = createHost(cases.simple);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should have css class accordion_panel__header', () => {
    subject = createHost(cases.simple);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should have css class accordion_panel__header--toggleanime', () => {
    subject = createHost(cases.expanded);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should have css class accordion_panel__header--hidetoggle', () => {
    subject = createHost(cases.hiddeToggle);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should render title', () => {
    subject = createHost(cases.title);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should render description', () => {
    subject = createHost(cases.description);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should render a custom icon', () => {
    subject = createHost(cases.customIcon);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should render title, description and icon together', () => {
    subject = createHost(cases.complex);
    expect(subject.hostFixture).toMatchSnapshot();
  });
});
