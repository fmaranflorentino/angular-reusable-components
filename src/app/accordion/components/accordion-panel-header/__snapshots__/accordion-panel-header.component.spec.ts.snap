// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionPanelHeaderComponent should create and match raw component snapshot 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        />
        
        <span
          class="material-icons toggle"
        >
           keyboard_arrow_down 
        </span>
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;

exports[`AccordionPanelHeaderComponent should have css class accordion_panel__header 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        />
        
        <span
          class="material-icons toggle"
        >
           keyboard_arrow_down 
        </span>
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;

exports[`AccordionPanelHeaderComponent should have css class accordion_panel__header--hidetoggle 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header accordion_panel__header--hidetoggle"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        />
        
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;

exports[`AccordionPanelHeaderComponent should have css class accordion_panel__header--toggleanime 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header accordion_panel__header--toggleanime"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        />
        
        <span
          class="material-icons toggle"
        >
           keyboard_arrow_down 
        </span>
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;

exports[`AccordionPanelHeaderComponent should render a custom icon 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        >
          
        </span>
        
        <span
          class="material-icons toggle"
        >
           keyboard_arrow_down 
        </span>
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;

exports[`AccordionPanelHeaderComponent should render description 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      <span
        paneldescription=""
      />
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        />
        
        <span
          class="material-icons toggle"
        >
           keyboard_arrow_down 
        </span>
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;

exports[`AccordionPanelHeaderComponent should render title 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        />
        
        <span
          class="material-icons toggle"
        >
           keyboard_arrow_down 
        </span>
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;

exports[`AccordionPanelHeaderComponent should render title, description and icon together 1`] = `
<ng-component>
  <app-accordion-panel-header
    class="accordion_panel__header"
  >
    <div
      class="accordion_panel__header__wrapper"
    >
      
      <span
        paneldescription=""
      />
      <div
        class="custom_icon__container"
      >
        <span
          class="custom_icon"
        >
          
        </span>
        
        <span
          class="material-icons toggle"
        >
           keyboard_arrow_down 
        </span>
      </div>
    </div>
  </app-accordion-panel-header>
</ng-component>
`;
